### PR TITLE
Make the Turret subsystem into a PID subsystem

### LINKED ISSUE #28: Make Turret A PIDSubsystem
Make the turret into a `PIDSubsystem` and change commands to take advantage of this rather than duplicating turret position control everywhere.

### PR DESCRIPTION
**Pull request by MatthewZ0823**
Fix #28 

Made the Turret subsystem into a PID subsystem

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Turret.java [hunk 1]
```diff
@@ -46,4 +53,15 @@ public void stop(){
...
+  @Override
+  public void useOutput(double output, double setpoint) {
+    // Use the output here
+  }
+
+  @Override
+  public double getMeasurement() {
+    // Return the process variable measurement here
+    return 0;
+  }
```
Comment (Jacob4649) [inline src/main/java/frc/robot/subsystems/Turret.java]: Functionality of the already existing move turret commands should be implemented here

**Review by Jacob4649 — CHANGES_REQUESTED** [review]
`setSetpoint` and `getSetpoint` methods should be implemented, can look at the general format of the `Shooter` class to see roughly how a `PIDSubsystem` should be implemented. Also make sure to remake existing turret commands so that they take advantage of the new `PIDSubsystem` functionality.

File: src/main/java/frc/robot/subsystems/Turret.java (hunk 1 above)
Comment (MatthewZ0823) [reply src/main/java/frc/robot/subsystems/Turret.java]: Fix in commit f018a89

**Comment by matthew-reynolds** [comment]
Closed since #28 has been closed in favor of #47.

Sorry to pull the rug out from under you! Thanks for your work, even if we aren't going to end up using it on the final robot.

**Pull request closed without merging**